python/cudf/dataframe/dataframe.py

```diff
@@ -890,6 +891,66 @@ def _n_largest_or_smallest(self, method, n, columns, keep):
                 df[k] = self[k].reset_index().take(new_positions)
         return df.set_index(self.index.take(new_positions))
 
+    def transpose(self):
+        """Transpose index and columns.
+
+        Returns
+        -------
+        a new (ncol x nrow) dataframe. self is (nrow x ncol)
+
+        Difference from pandas
+        ----------------------
+        Not supporting *copy* because default and only behaviour is copy=True
+        """
+        dtype = self.dtypes[0]
+        if any(t != dtype for t in self.dtypes):
+            raise ValueError('all columns must have the same dtype')
+        has_null = any(c.null_count for c in self._cols.values())
+
+        df = DataFrame()
+
+        ncols = len(self.columns)
+        cols = [self[col]._column.cffi_view for col in self._cols]
+
+        new_nrow = ncols
+        new_ncol = len(self)
+
+        if has_null:
+            new_col_series = [
+                Series.from_masked_array(
+                    data=Buffer(rmm.device_array(shape=new_nrow, dtype=dtype)),
+                    mask=cudautils.make_empty_mask(size=new_nrow),
```

Comment:
Do you care about setting the mask to all zeros here? Could you just initialize the mask without setting the bits for perf?

Reply:
In cudf_table.cuh:487 the valids buffer is sent to the c++ transpose function which assumes the data is 0 init. because it sets bits with an OR